showcase/src/elm/Utils/FileParser.elm

```diff
@@ -0,0 +1,389 @@
+module Utils.FileParser exposing (elliefy)
+
+
+{- | File-related Logic
+-}
+
+import Parser exposing ((|.), (|=), DeadEnd, Parser, Problem(..), Step(..))
+import Url.Builder
+
+
+
+-------------------
+-- PARSING STUFF
+-- Elliefy makes the code renderable in Ellie. WIP
+-------------------
+
+
+elliefy : String -> { title : String, code : String }
+elliefy srcCode =
+    let
+        ast =
+            Parser.run fileParser srcCode
+
+        title =
+            case ast of
+                Ok file ->
+                    getTitle file
+
+                Err _ ->
+                    "Parsing Failure"
+
+        parsed =
+            Result.map (changeModuleNameTo "Main") ast
+                |> Result.map elliefyExports
+                |> Result.map toString
+
+        parsedSrcCode =
+            case parsed of
+                Err parserErr ->
+                    -- elm code will literally just contain the error message if parsing fails
+                    "Error! " ++ deadEndsToString parserErr
+
+                Ok code ->
+                    code
+    in
+    { title = title
+    , code = parsedSrcCode
+    }
+
+
+
+-------------------
+-- DATA
+-------------------
+
+
+type alias File =
+    { moduleDeclaration : ModuleDeclaration
+    , imports : String
+    , codeBlocks : List CodeBlock
+    }
+
+
+type alias ModuleDeclaration =
+    { name : String
+    , exposes : List String
+    }
+
+
+type CodeBlock
+    = TypeDef String
+    | FunDef FunInfo
+
+
+type alias FunInfo =
+    { funName : String
+    , typeAnnotation : String
+
+    -- everything after the equals sign
+    , val : String
+    }
+
+
+
+-------------------
+-- PARSERS
+-------------------
+
+
+fileParser : Parser File
+fileParser =
+    Parser.succeed File
+        |= moduleDeclarationParser
+        |. spacesAndComments
+        |= imports
+        |. spacesAndComments
+        |= codeBlocks
+
+
+moduleDeclarationParser : Parser ModuleDeclaration
+moduleDeclarationParser =
+    let
+        moduleName : Parser String
+        moduleName =
+            Parser.chompUntil " "
+                |> Parser.getChompedString
+    in
+    Parser.succeed ModuleDeclaration
+        |. Parser.keyword "module"
+        |. Parser.spaces
+        |= moduleName
+        |. Parser.spaces
+        |. Parser.keyword "exposing"
+        |. Parser.spaces
+        |= exposes
```

Comment:
General observation:

You have two naming conventions:

- `<parserName>Parser`
- `<parserName` 
  - i.e. you don't explicitly say that it's a parser

Not a big deal since these functions aren't exposed, but it's an inconsistency nonetheless :)